src/V3Case.cpp

```diff
@@ -227,15 +228,51 @@ class CaseVisitor final : public VNVisitor {
                 }
             }
         }
-        for (uint32_t i = 0; i < numCases; ++i) {
-            if (!m_valueItem[i]) {
-                nodep->v3warn(CASEINCOMPLETE, "Case values incompletely covered "
-                                              "(example pattern 0x"
-                                                  << std::hex << i << ")");
-                m_caseNoOverlapsAllCovered = false;
-                return false;
+        if (AstEnumDType* enumDtp = VN_CAST(nodep->exprp()->dtypep()->skipRefToEnump(), EnumDType)) { // CASE is enum
+            if (nodep->uniquePragma() || nodep->priorityPragma()) {
+              suppressCaseIncWarn =true; // suppress the warning when we have unique or priority
+            } else { // check if all values were supplied
+                 bool foundMatch =false;
+                 for (uint32_t i = 0; i < numCases; ++i) { // iterate through all case items
+                     for (AstEnumItem* itemp = enumDtp->itemsp(); itemp;
+                                       itemp = VN_AS(itemp->nextp(), EnumItem)) { //iterate through all enum values
+                          if ( m_valueItem[i])  {
+                           AstConst* const constp = VN_AS(itemp->valuep(), Const);
+                           AstConst* const itemvp  = VN_AS(m_valueItem[i]->op1p(),Const);
+                             if (itemvp->num().toDecimalU().compare(constp->num().toDecimalU()) == 0 ) { // match
+                                 foundMatch= true;
```

Comment:
```suggestion
                             if (itemvp->num().isCaseEq(constp->num())) {
                                 foundMatch = true;
```

Reply:
Fixed in 4a416e0ee4160bd57d938c6cd0d470c20a428f37